src/Carousel/Carousel.tsx

```diff
@@ -1,64 +1,140 @@
-import * as React from "react";
-import Slider, { Settings } from "react-slick";
-import "../../node_modules/slick-carousel/slick/slick.css";
-import "../../node_modules/slick-carousel/slick/slick-theme.css";
-import "./carousel-style.scss";
+import React from "react";
+import classnames from "classnames";
+import { randomId } from "@sebgroup/frontend-tools/dist/randomId";
+import { CarouselItem, CarouselItemProps } from "./CarouselItem";
+import { CarouselIndicators } from "./CarouselIndicators";
+import { CarouselNavs } from "./CarouselNavs";
+import "./carousel.scss";
 
-export interface CarouselItem {
-    title?: string;
-    desc?: string;
-    image?: string;
-}
-
-export interface CarouselProps {
+export interface CarouselProps extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
+    /** Event handler triggered after change have happened to the carousel returning the index of the new active carousel slide */
     afterChange?: (index: number) => void;
-    autoPlay?: boolean;
-    autoPlaySpeed?: number;
-    backgroundPlacement?: string;
-    className?: string;
-    height?: number;
-    id?: string;
+    /** The duration (in milliseconds) it takes to transition from one slide to another. Default is `600` */
+    transitionDuration?: number;
+    /** Transition style. Supported styles: `slide` and `fade` */
+    transitionStyle?: "slide" | "fade";
+    /** Enables infinite scrolling */
     infinite?: boolean;
-    list: Array<CarouselItem>;
+    /** Shows clickable indicators at the bottom */
+    showIndicators?: boolean;
+    /** A list of accordion items to be rendered */
+    list?: Array<CarouselItemProps>;
+    /** The index of default active slide */
+    defaultValue?: number;
```

Comment:
i think there's better naming for this, defaultValue is a bit confusing

Reply:
It’s a native HTML attribute, I’m just reusing it. If all components share HTML native attributes we will have more consistent behavior don’t you think?